java/statestore/src/main/java/sleeper/statestore/transactionlog/TransactionLogFilesSnapshot.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.statestore.transactionlog;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.hadoop.ParquetReader;
+import org.apache.parquet.hadoop.ParquetWriter;
+
+import sleeper.core.record.Record;
+import sleeper.core.schema.Field;
+import sleeper.core.schema.Schema;
+import sleeper.core.schema.type.IntType;
+import sleeper.core.schema.type.LongType;
+import sleeper.core.schema.type.StringType;
+import sleeper.core.statestore.AllReferencesToAFile;
+import sleeper.core.statestore.FileReference;
+import sleeper.core.statestore.FileReferenceSerDe;
+import sleeper.core.statestore.StateStoreException;
+import sleeper.core.statestore.transactionlog.StateStoreFiles;
+import sleeper.io.parquet.record.ParquetReaderIterator;
+import sleeper.io.parquet.record.ParquetRecordReader;
+import sleeper.io.parquet.record.ParquetRecordWriterFactory;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Instant;
+import java.util.List;
+
+public class TransactionLogFilesSnapshot {
+    private static final Schema FILE_SCHEMA = initialiseFilesSchema();
+    private final Configuration configuration;
+    private final FileReferenceSerDe serDe = new FileReferenceSerDe();
+
+    TransactionLogFilesSnapshot(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    void save(java.nio.file.Path tempDir, StateStoreFiles state, long lastTransactionNumber) throws StateStoreException {
```

Comment:
The path parameters to these methods are all called tempDir, and they're Java file system paths instead of Hadoop paths or strings. They can probably just be strings? It looks like it's supposed to be the path to the state store files for the Sleeper table?